ls-ui-react-app/src/apiResponses.ts

```diff
@@ -0,0 +1,51 @@
+import {Slices} from "./slices";
+
+// Different classes representing possible Api Response outcomes
+class JsonParseError {
+  msg: string
+  constructor(msg : string) {
+    this.msg = msg
+  }
+}
+
+class ApiHttpError {
+  code: number
+  msg: string
+  headers: string
+  constructor(code: number, msg : string, headers: string) {
+    this.code = code
+    this.msg = msg
+    this.headers = headers
+  }
+}
+
+class CompilerError {
+  msg: string
+  errortype: string
+  constructor(msg: string, errortype: string){
+    this.msg = msg
+    this.errortype = errortype;
+  }
+}
+
+class CompilationResultSuccess {
+  slices: Slices
+  compilationText: string
+  constructor(slices : Slices, compilationText : string) {
+    this.slices = slices
+    this.compilationText = compilationText
+  }
+}
+
+class NoServerResponse {
+  response: string
```

Comment:
I'm just not sure ... why is there a string here if the type is no response?

Reply:
The string is developer-defined to be whatever we want the "no server response" error message to say. We don't have to assign it to this class, we could just render some arbitrary text in UploadCircuitPage rather than assigning it to a class.